flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/TieredStorageProducerClient.java

```diff
@@ -80,13 +101,18 @@ public void write(
 
         if (isBroadcast && !isBroadcastOnly) {
             for (int i = 0; i < numSubpartitions; ++i) {
-                bufferAccumulator.receive(record.duplicate(), subpartitionId, dataType);
+                bufferAccumulator.receive(
+                        record.duplicate(), new TieredStorageSubpartitionId(i), dataType);
```

Comment:
Let's fix this in a separate hotfix commit.

Reply:
OK, Fixed it with a separate commit.